### ISSUE TITLE
`SystemExit: Could not compile Unet.` every time on commit `206c1b7`

### ISSUE THREAD

**Issue by xzuyn** [nod-ai/SHARK#1704]
- Commit: `206c1b7`
- Flags: `--local_tank_cache=D:\Diffusion_Stable\SHARK_CACHE --output_dir="Z:\Windows Directories\Downloads\Shark_Images" --no-progress_bar --device_allocator=caching`
- GPU: `RX 7900 XTX 24GB`
- CPU: `i5-10400f`
- RAM: `32GB 2800MHz`
- Drivers: `23.7.2`
- OS: `Windows 10 Pro (Version 10.0.19045 Build 19045)`
- AMD_ENABLE_LLPC: `1`

I always get this error when I go to run SHARK. It fails before making an image, then when trying again it re-downloads the model and still fails.

This is when I first ran: [shark_error_first-clean-run.txt](https://github.com/nod-ai/SHARK/files/12209974/shark_error_first-clean-run.txt)
This is retrying: [shark_error_rerunning-without-deleting-anything.txt](https://github.com/nod-ai/SHARK/files/12210002/shark_error_rerunning-without-deleting-anything.txt)

Tried resetting the venv, deleting the `*.vmfb` files, latest commit, latest drivers. Last known commit that I know works (besides an issue I have with `clear_cache` closing SHARK) is one from `9fcae4f` cause its the latest I had used for hires fix. I need to bisect and find which commit breaks things for me.

**Comment by xzuyn** [comment]
I tried bisecting (`good 9fcae4f` and `bad 206c1b7`), but all the older ones just failed with this when trying to setup the venv.
```
ERROR: Could not find a version that satisfies the requirement torch-mlir==20230620.875 (from nodai-shark) (from versions: 20230703.888, 20230704.889, 20230705.890, 20230706.891, 20230707.892, 20230708.893, 20230709.894, 20230710.89
5, 20230711.896, 20230712.897, 20230713.898, 20230714.899, 20230715.900, 20230716.901, 20230717.902, 20230718.903, 20230719.118, 20230719.119, 20230719.904, 20230720.905, 20230721.906, 20230722.907, 20230723.908, 20230724.909, 20230
725.910, 20230726.911, 20230727.912, 20230728.913, 20230729.914, 20230730.915)
ERROR: No matching distribution found for torch-mlir==20230620.875
```

**Comment by NeedsMoar** [comment]
I hit this issue too, seemed to start at 840.  835 had a 3it/s performance regression so I've been using the released version.  

The part that makes me suspicious is the cooperative matrix extension in the target triple;  that didn't exist until the July 6th drivers which shark previously wasn't compatible with (when tuning was on) so I moved to the late June beta agility SDK drivers with hardware scheduling support thinking I might mess with that (I didn't).  Then I found out that OpenCL was broken in those while trying to use vceencc64 and failing suddenly and reverted to the last release version... *sigh*  I'm going to try the July drivers again and see if they're now a requirement.  I don't know how the drivers would be related to a compilation error but the compilation pipeline for shark is weird so who knows.  

Incidentally I tried --no-use_tuned and got the same error as with tuned files so it's more complicated than that.

**Comment by NeedsMoar** [comment]
Driver version didn't matter, as expected.   Regression somewhere between 819 and 835 and drops tuned generation at 512x768 from close to 14it/s to 10it/s so it's a bit bigger than I thought.  I can't test anything newer than that because of the compilation error here.

**Comment by xzuyn** [comment]
Solved now. Something fixed it between `206c1b7` & `872bd72`.

@NeedsMoar try out the latest, it may work for you. `50it [00:02, 21.46it/s]` on with the default settings on a 7900XTX.
